src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,152 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.duke.Duke;
+import duke.exceptions.InvalidArgumentException;
+import duke.exceptions.InvalidCommandException;
+import duke.utils.DateValidator;
+import duke.utils.DateValidatorUsingDateFormat;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input into a command.
+     * @param userInput User input string.
+     * @param bot A Duke object that manages task list operations.
+     * @return User command.
+     * @throws InvalidCommandException If user input is an unrecognised command.
+     * @throws InvalidArgumentException If argument is missing from user input.
+     */
+    public static Command processInput(String userInput, Duke bot) throws InvalidCommandException,
```

Comment:
Maybe could have abstract each if-else statement into functions to increase readability.